cfme/tests/containers/test_chargeback_for_images.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+
+import fauxfactory
+from random import random
+
+import pytest
+
+from utils import testgen
+from utils.log import logger
+from utils.providers import list_providers_by_class
+from cfme.containers.provider import ContainersProvider
+from cfme.intelligence import chargeback as cb
+from cfme.intelligence.reports.reports import CustomReport
+
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.long_running,
+    pytest.mark.meta(
+        server_roles='+ems_metrics_coordinator +ems_metrics_collector +ems_metrics_processor'),
+    pytest.mark.usefixtures('setup_provider')
+]
+pytest_generate_tests = testgen.generate([ContainersProvider], scope='function')
+
+
+def new_chargeback_rate(appliance, include_variable_rates=True):
+
+    # Create a new Chargeback compute rate
+
+    def rand_float_str():
+        return str(round(random() * fauxfactory.gen_integer(1, 20), 2))
+
+    def gen_var_rate():
+        return (rand_float_str() if include_variable_rates else 0)
+
+    description = 'custom_rate_' + fauxfactory.gen_alphanumeric()
+    data = {
+        'Used CPU Cores': {'per_time': 'Hourly',
+                           'fixed_rate': fauxfactory.gen_integer(1, 4),
+                           'variable_rate': gen_var_rate()},
+        'Fixed Compute Cost 1': {'per_time': 'Hourly',
+                                 'fixed_rate': rand_float_str()},
+        'Fixed Compute Cost 2': {'per_time': 'Hourly',
+                                 'fixed_rate': rand_float_str()},
+        'Used Memory': {'per_time': 'Hourly',
+                        'fixed_rate': rand_float_str(),
+                        'variable_rate': gen_var_rate()},
+        'Used Network I/O': {'per_time': 'Hourly',
+                             'fixed_rate': rand_float_str(),
+                             'variable_rate': gen_var_rate()}
+    }
+    ccb = cb.ComputeRate(description, fields=data, appliance=appliance)
+    ccb.create()
+    return ccb
+
+
+@pytest.fixture(scope='module')
+def provider_name():
+    # Since 'provider' fixture is a 'function' scoped, we use this
+    # fixture to yield the provider name
```

Comment:
I believe you can use ``pytest.mark.usefixtures("setup_provider_modscope")`` (instead of "setup_provider")